### DISCUSSION TITLE
Feature Request: Durable Objects Storage should have a way to list keys without retrieving values

### DISCUSSION THREAD

**Discussion by radix** [cloudflare/workerd#1319] [Ideas]
There is a list() function which returns keys and values.

Request: There should be a list_keys() function that just returns the keys.

For example, I have a DO which stores its data in a small number of snapshots - say, less than a thousand for the lifetime of the DO; essentially a trivial number of rows to query out of SQLite. I store a key like `"snapshot-{index}"`, and the value is large (close to the 128kB limit). When I need to write a new snapshot, I would like to just list the snapshot keys to see what the next index is, and then write to that index. Since the values are large, I don't want to load them into memory, so I don't want to use list().

Of course, I can maintain and synchronize the index in a separate key, but obviously just being able to list the keys would be a simpler solution that would prevent the possibility of the data getting out of sync.

**Comment by a-robinson** [comment]
@joshthoward - this is a relatively common request
